### PR TITLE
Add PUT for updating reactive tab settings

### PR DESCRIPTION
**Pull request by paniagua**
@deiwin @indrekj  can you guys please take a look?

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
...
+* `placement`       *It places the reactive to the 'right' or 'left' side of the browser*
+* `vertical_offset` *It sets the position of the reactive bar vertically. It is a integer from '0' to '100', where '100' is the top of the browser*
+* `format`      *It defines if the reactive tab contains just 'text' or it is a 'mix' between text and icons*
+* `size`        *It is a number 80 to 100 that sets the size of the reactive tab*
+* `text`        *The text displayed in the reactive tab*
+* `font_size`   *The size of the font used to display the reactive tab text. Available font sizes are: 20, 24, 27*
+* `icon_class`  *The icon to be shown in the reactive tab. Available icons are: 'ico-voice', 'ico-text', 'ico-video'*
+* `front_color` *Front color of the reactive tab in hexa format*
+* `back_color`  *Back color of the reactive tab in hexa format*
+* `hover_color` *Hover color of the reactive tab in hexa format*
```
Comment (stenver) [inline apiary.apib]: Why are the explanations in Italic? Maybe a table would be clearer here

File: apiary.apib [hunk 2]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
...
             }
 
+## Reactive Tab [/sites/{site_id}/reactive_tab]
+
+Updates a site's reactive tab configuration. The operator making the request should have privileges to do so.
+
+* `placement`       *It places the reactive to the 'right' or 'left' side of the browser*
+* `vertical_offset` *It sets the position of the reactive bar vertically. It is a integer from '0' to '100', where '100' is the top of the browser*
+* `format`      *It defines if the reactive tab contains just 'text' or it is a 'mix' between text and icons*
+* `size`        *It is a number 80 to 100 that sets the size of the reactive tab*
```
Comment (stenver) [inline apiary.apib]: Can we explain this? Why is it from 80 to 100?

File: apiary.apib (hunk 1 above)
Comment (stenver) [inline apiary.apib]: Would be good to specify the available options for some enums. Maybe add separate column for explaining possible values and type

File: apiary.apib [hunk 3]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
               "hostnames": ["some.domain.com", "other.domain.com"]
             }
 
+## Reactive Tab [/sites/{site_id}/reactive_tab]
+
+Updates a site's reactive tab configuration. The operator making the request should have privileges to do so.
+
+* `placement`       *It places the reactive to the 'right' or 'left' side of the browser*
+* `vertical_offset` *It sets the position of the reactive bar vertically. It is a integer from '0' to '100', where '100' is the top of the browser*
+* `format`      *It defines if the reactive tab contains just 'text' or it is a 'mix' between text and icons*
```
Comment (stenver) [inline apiary.apib]: This could use a `available_values` column or such

File: apiary.apib [hunk 4]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
...
+* `size`        *It is a number 80 to 100 that sets the size of the reactive tab*
+* `text`        *The text displayed in the reactive tab*
+* `font_size`   *The size of the font used to display the reactive tab text. Available font sizes are: 20, 24, 27*
+* `icon_class`  *The icon to be shown in the reactive tab. Available icons are: 'ico-voice', 'ico-text', 'ico-video'*
+* `front_color` *Front color of the reactive tab in hexa format*
```
Comment (stenver) [inline apiary.apib]: `hex` format?

File: apiary.apib (hunk 1 above)
Comment (stenver) [inline apiary.apib]: It would be good if we had a consistent way to do this @deiwin

File: apiary.apib [hunk 5]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
               "hostnames": ["some.domain.com", "other.domain.com"]
             }
 
+## Reactive Tab [/sites/{site_id}/reactive_tab]
+
+Updates a site's reactive tab configuration. The operator making the request should have privileges to do so.
```
Comment (stenver) [inline apiary.apib]: What privileges are required? Since we don't have privilege based system, it should probably be manager

File: apiary.apib [hunk 6]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
...
+
+* `placement`       *It places the reactive to the 'right' or 'left' side of the browser*
+* `vertical_offset` *It sets the position of the reactive bar vertically. It is a integer from '0' to '100', where '100' is the top of the browser*
```
Comment (deiwin) [inline apiary.apib]: `bar` -> `tab`. Also, here and elsewhere in the list, capitalize "Reactive Tab".

File: apiary.apib [hunk 7]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
               "hostnames": ["some.domain.com", "other.domain.com"]
             }
 
+## Reactive Tab [/sites/{site_id}/reactive_tab]
+
+Updates a site's reactive tab configuration. The operator making the request should have privileges to do so.
+
+* `placement`       *It places the reactive to the 'right' or 'left' side of the browser*
```
Comment (deiwin) [inline apiary.apib]: `reactive` -> `Reactive Tab`

File: apiary.apib (hunk 5 above)
Comment (deiwin) [inline apiary.apib]: This text should be under the `PUT` section

File: apiary.apib (hunk 7 above)
Comment (deiwin) [inline apiary.apib]: All of these arguments should be in the `PUT` section and described under `Attributes`, as done [here](https://github.com/salemove/apiary_docs/pull/27/files#diff-2625016b50d68d922257f74801cac29cR1365)

File: apiary.apib (hunk 4 above)
Comment (paniagua) [inline apiary.apib]: spanish slipped in :P

**Comment by deiwin** [comment]
Can you please include a screenshot of how the `members` contents will be rendered?

**Comment by paniagua** [comment]
Actually, i am wrong, cant use the member thing in the attributes. fixing.

**Comment by paniagua** [comment]
Using Parameters with Members

Using Parameters with members 

Using Attributes Only
Using Attributes only

**Comment by paniagua** [comment]
Are we ok with the param names? i would like to start doing the changes

File: apiary.apib [hunk 8]
```diff
@@ -1653,7 +1653,7 @@ Updates a site's configuration. The operator making the request should have priv
 
 Updates a site's Reactive Tab configuration. The operator making the request should have privileges to do so.
 
-+ Parameters
++ Parametersdd
```
Comment (deiwin) [inline apiary.apib]: dd?

Comment (deiwin) [inline apiary.apib]: vimtrails?

File: apiary.apib [hunk 9]
```diff
@@ -1647,6 +1647,92 @@ Updates a site's configuration. The operator making the request should have priv
...
+    + site_id (required, number, `1`) ... ID of the site in form of an integer
+
++ Request
+
+    + Headers
+
+            Authorization: Token 2pFuPlTYkH7T4MylrwxApA
+            Accept: application/vnd.salemove.v1+json
+
+    + Parameters
```
Comment (deiwin) [inline apiary.apib]: What's the difference between `Parameters` and `Attributes`?  I thought that `Parameter` is only for the parameters specified in the URL, e.g. `site_id` in `/sites/{site_id}/reactive_tab`.

**Comment by stenver** [comment]
Nice upgrade to explain request body attributes (Y)

File: apiary.apib [hunk 10]
```diff
@@ -1647,6 +1647,92 @@ Updates a site's configuration. The operator making the request should have priv
               "hostnames": ["some.domain.com", "other.domain.com"]
             }
 
+## Reactive Tab [/sites/{site_id}/reactive_tab]
+
+### Update Reactive Tab [PUT]
+
+Updates a site's Reactive Tab configuration. The operator making the request should have privileges to do so.
+
++ Parametersdd
```
Comment (stenver) [inline apiary.apib]: `Parametersdd`

And below should probably be `Attributes`, as @deiwin mentioned

File: apiary.apib (hunk 9 above)
Comment (paniagua) [inline apiary.apib]: Apiary uses the `Parameters` when building a request in the `sandbox`. `Attributes` can be used in the body. I set all as `Attributes`. You can't use the `Members` clause in the `Attributes`.

**Comment by stenver** [comment]
+1

File: apiary.apib [hunk 11]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
               "hostnames": ["some.domain.com", "other.domain.com"]
             }
 
+## Reactive Tab [/sites/{site_id}/reactive_tab]
+
+### Update Reactive Tab [PUT]
+
+Updates a site's Reactive Tab configuration. The operator making the request should have privileges to do so.
```
Comment (deiwin) [inline apiary.apib]: Stenver's comment about what those privileges are seems to have been forgotten.

File: apiary.apib [hunk 12]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
...
+
++ Request
+
+    + Headers
+
+            Authorization: Token 2pFuPlTYkH7T4MylrwxApA
+            Accept: application/vnd.salemove.v1+json
+
+    + Attributes
+      + placement (string) - It places the Reactive Tab to the 'right' or 'left' side of the browser. Possible values `right`, `left`
```
Comment (deiwin) [inline apiary.apib]: The `It ...` sentences [read weird](https://cuppacriterion.files.wordpress.com/2014/12/silence17.jpg). Maybe `Onto which side of the browser to place the Reactive Tab. Possible values ...`

File: apiary.apib [hunk 13]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
...
+      + vertical_offset (number) - It sets the position of the Reactive Tab vertically. It is a integer from '0' to '100', where '100' is the top of the browser
```
Comment (deiwin) [inline apiary.apib]: `The vertical offset of the Reactive Tab from the bottom of the browsers viewport, expressed as a percentage. Possible values range from '0' to '100'.`

File: apiary.apib [hunk 14]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
...
+      + format (string) - It defines if the Reactive Tab contains just 'text' or it is a 'mix' between text and icons. Possible values `text`, `mix`
```
Comment (deiwin) [inline apiary.apib]: Whether the Reactive Tab should contain just `text` or a `mix` of both text and icons. Possible values ...

File: apiary.apib [hunk 15]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
...
+      + size (number) - It is a number 80 to 100 that sets the size of the Reactive Tab. Any value outside (80, 100) impacts the look & feel of the Reactive Tab
```
Comment (deiwin) [inline apiary.apib]:
```
The scale of the Reactive Tab as a percentage of the default size. Suggested values are between `80` and `100`.

File: apiary.apib [hunk 16]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
...
+    + Headers
+
+            Authorization: Token 2pFuPlTYkH7T4MylrwxApA
+            Accept: application/vnd.salemove.v1+json
+
+    + Attributes
+      + placement (string) - Onto which side of the browser to place the Reactive Tab. Possible values ```right```, ```left```.
+      + vertical_offset (number) - The vertical offset of the Reactive Tab from the bottom of the browsers viewport, expressed as a percentage. Possible values range from ```0``` to ```100```.
+      + format (string) - Whether the Reactive Tab should contain just ```text``` or a ```mix``` of both text and icons. Possible values ```text```, ```mix```.
+      + size (number) - The scale of the Reactive Tab as a percentage of the default size. Suggested values are between ```80``` and ```100```.
```
Comment (deiwin) [inline apiary.apib]: I think the triple ticks (```) will put everything in them onto separate lines, no?  I suggest you change to use only single ticks here and above.

File: apiary.apib [hunk 17]
```diff
@@ -1647,6 +1647,64 @@ Updates a site's configuration. The operator making the request should have priv
...
++ Request
+
+    + Headers
+
+            Authorization: Token 2pFuPlTYkH7T4MylrwxApA
+            Accept: application/vnd.salemove.v1+json
+
+    + Attributes
+      + placement (string) - Onto which side of the browser to place the Reactive Tab. Possible values ```right```, ```left```.
+      + vertical_offset (number) - The vertical offset of the Reactive Tab from the bottom of the browsers viewport, expressed as a percentage. Possible values range from ```0``` to ```100```.
```
Comment (deiwin) [inline apiary.apib]: `browsers` -> `browser's`

**Comment by deiwin** [comment]
2 quibbles, +1 otherwise

**Comment by stenver** [comment]
+1 after Deiwins comments

**Comment by paniagua** [comment]
Updated

**Comment by deiwin** [comment]
+1

**Comment by paniagua** [comment]
Got +1 from Stenver on chat - merging

**Comment by paniagua** [comment]
!squash

**Pull request merged**